sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -106,14 +106,14 @@ class SparkSqlAstBuilder extends AstBuilder {
         SetCommand(Some(keyStr -> None))
       }
     } else {
-      remainder(ctx.SET.getSymbol).trim match {
+      remainder(ctx.SET.getSymbol).trim.replaceAll(";+$", "") match {
```

Comment:
is it the same as `.trimRight(";")`?

Reply:
Well, `.trimRight(";")` can only be done on `UTF8String`, but here we have java `String`. In java we do not have formal trimRight or trimLeft, but only trim for spaces. This is a regex replace to exchange all occurrences of `;` at the end ($) of the String.